src/KafkaFlow/Producers/MessageProducer.cs

```diff
@@ -44,20 +44,37 @@ internal class MessageProducer : IMessageProducer, IDisposable
 
             DeliveryResult<byte[], byte[]> report = null;
 
-            await this.middlewareExecutor
-                .Execute(
-                    new ProducerMessageContext(
-                        message,
-                        messageKey,
-                        headers,
-                        topic),
-                    async context =>
-                    {
-                        report = await this
-                            .InternalProduceAsync((ProducerMessageContext) context)
-                            .ConfigureAwait(false);
-                    })
-                .ConfigureAwait(false);
+            try
+            {
+                await this.middlewareExecutor
+                    .Execute(
+                        new ProducerMessageContext(
+                            message,
+                            messageKey,
+                            headers,
+                            topic),
+                        async context =>
+                        {
+                            report = await this
+                                .InternalProduceAsync((ProducerMessageContext) context)
+                                .ConfigureAwait(false);
+                        })
+                    .ConfigureAwait(false);
+            }
+            catch (Exception ex) when (!(ex is ProduceException<byte[], byte[]> || ex is KafkaException))
+            {
+                this.dependencyResolverScope.Resolver
+                    .Resolve<ILogHandler>()
+                    .Error("Error executing producer",
+                        ex,
+                        new
+                        {
+                            message,
+                            topic,
+                            partitionKey
+                        });
+                throw;
+            }
```

Comment:
I think we should not change this behaviour. The app is responsible to catch the exceptions and choose whats to do in these cases.

Reply:
done